### ISSUE TITLE
"The bless of the installer disk image failed." For Mavericks installer

### ISSUE THREAD

**Issue by mihirpotnis** [BITespresso/createinstalliso#7]
So when I run  "sudo ~/bin/createinstalliso --isodirectory ~/Desktop/ --applicationpath ~/Desktop/Install\ OS\ X\ Mavericks.app/", I get:

Making disk image bootable...
The bless of the installer disk image failed.

I tried "codesign --remove-signature ~/bin/createInstalliso" and also running through rosetta to no avail. 

It probably is the certification error but how do I get past it?

**Comment by despin** [comment]
I am also trying to debug this particular error for the Mavericks installer. 

Modifying the script to output the complete bless command, and executing it outside of the script like this: 
```
bless --folder {random folder given by the script}/InstallDisk/System/Library/CoreServices --file {random folder given by the script}/InstallDisk/System/Library/CoreServices/boot.efi --openfolder {random folder given by the script}/InstallDisk --label Install OS X Mavericks
```
Outputs the following error: `bless: The 'openfolder' is not supported on Apple Silicon devices.` 

After some digging, apparently this option is not needed for M1 devices. [Evidence](https://github.com/LinusU/node-appdmg/pull/223). So I just removed the option from the script.

Try that. I'm using this script to run Mavericks on UTM, with utmconfigs, but I have [this exact issue](https://github.com/adespoton/utmconfigs/issues/7). Maybe your use case is different and the solution specified earlier might help you.

**Comment by MrGlitchWasTaken** [comment]
> I am also trying to debug this particular error for the Mavericks installer.
> 
> Modifying the script to output the complete bless command, and executing it outside of the script like this:
> 
> ```
> bless --folder {random folder given by the script}/InstallDisk/System/Library/CoreServices --file {random folder given by the script}/InstallDisk/System/Library/CoreServices/boot.efi --openfolder {random folder given by the script}/InstallDisk --label Install OS X Mavericks
> ```
> 
> Outputs the following error: `bless: The 'openfolder' is not supported on Apple Silicon devices.`
> 
> After some digging, apparently this option is not needed for M1 devices. [Evidence](https://github.com/LinusU/node-appdmg/pull/223). So I just removed the option from the script.
> 
> Try that. I'm using this script to run Mavericks on UTM, with utmconfigs, but I have [this exact issue](https://github.com/adespoton/utmconfigs/issues/7). Maybe your use case is different and the solution specified earlier might help you.

any updates?

**Comment by mihirpotnis** [comment]
So the best way to get past this problem if I remember correctly is to use Mac OS X Lion on UTM, getting the mavericks installer on the virtual machine and running the bless command which will work as expected.

Running Lion on UTM was quite easy if I remember correctly (even snow leopard or leopard will work) and you can get the installer on there using an external usb drive and connecting it to the vm, then transferring your bootable image onto the USB drive again and accessing it on your Mac.

Unlike windows, Apple is quite terrible at backwards compatibility and they simply do not care either so.

**Comment by MrGlitchWasTaken** [comment]
> So the best way to get past this problem if I remember correctly is to use Mac OS X Lion on UTM, getting the mavericks installer on the virtual machine and running the bless command which will work as expected.
> 
> Running Lion on UTM was quite easy if I remember correctly (even snow leopard or leopard will work) and you can get the installer on there using an external usb drive and connecting it to the vm, then transferring your bootable image onto the USB drive again and accessing it on your Mac.
> 
> Unlike windows, Apple is quite terrible at backwards compatibility and they simply do not care either so.

by the way, do you know a fix for my mavericks iso where terminal says "Install OS X Mavericks.app/ does not appear to be a valid OS installer application."

**Comment by MrGlitchWasTaken** [comment]
> It is most likely due to expired certification and/or modern Mac OS actually can't recognize it as a valid Mac OS installer. I am pretty sure using a Mac OS older than the one you're trying to make an image for will fix the issue!

would doing this method work in a VM?

**Comment by mihirpotnis** [comment]
Yup! You can use UTM to install Lion and use bless in that VM!

**Comment by mihirpotnis** [comment]
Try it? Might be a bit too old but you never know

**Comment by MrGlitchWasTaken** [comment]
> Try it? Might be a bit too old but you never know

im using utm, how would i get the install os x file to the vm and i dont have internet connection for the vm

**Comment by mihirpotnis** [comment]
you can use an external usb and connect it to the VM to transfer data between the machines

**Comment by MrGlitchWasTaken** [comment]
> you can use an external usb and connect it to the VM to transfer data between the machines

omg that would have been useful to know a long time ago thanks!

**Comment by MrGlitchWasTaken** [comment]
> you can use an external usb and connect it to the VM to transfer data between the machines

do you know how to bypass the error "to download and restore OS X, your computer's eligibility will be verified with apple"

**Comment by MrGlitchWasTaken** [comment]
im on an OS X 10.7 vm

**Comment by BITespresso** [comment]
Solved in [v1.1.0](https://github.com/BITespresso/createinstalliso/releases/tag/v1.1.0)
